specification/communication/data-plane/CallingServer/preview/2021-06-15-preview/communicationservicescallingserver.json

```diff
@@ -0,0 +1,1467 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-06-15-preview",
+    "title": "Azure Communication Calling Server Service",
+    "description": "Azure Communication Calling Server Service"
+  },
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/calling/callConnections": {
+      "post": {
+        "tags": [
+          "CallConnection"
+        ],
+        "summary": "Create a new call.",
+        "description": "Create a new call.",
+        "operationId": "CallConnection_CreateCall",
```

Comment:
The first part of the operationId (i.e. `CallConnection`) becomes the name of the client exposing APIs, it is recommended to use the plural name `CallConnections`.

Reply:
As we determine the right taxonomy for call/connection etc, we should make sure that the API pattern is:

POSTin to `/calling/<term for callConnections in plural>` responds with the `Location` (standard header) to the newly created `/calling/<term for callConnections in plural>/<generated "name" of the created callConnection>`. The name of the model at that location should be <term for callConnections in singular>. 

E.g.
`
POST /calling/callConnections -> (Location of created CallConnection)
GET /calling/callConnections/{callConnectionName} -> CallConnection
`

Reply:
We use a guid to identify the callConnection and call it callConnectionId. Is this what you are referring to as callConnectionName or is that different? Currently the way it works is:
`POST /calling/callConnections` returns callConnectionId in the body of the 201 response. The app can do `GET /calling/callConnections/callConnectionId` Are you suggesting we should return the URL in the response?

Reply:
`id` is generally a fully qualified URL, which is why I suggested a different name for the "last segment". 

And, yes, for a POST/201 response, I would expect a [Location header](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.2) indicating where the resource got created.